ui/src/common/components/CitationSummaryGraph/CitationSummaryGraph.jsx

```diff
@@ -11,63 +11,157 @@ import {
 
 const LEGENDS = [
   { title: 'Citeable', color: BLUE },
   { title: 'Published', color: ORANGE },
 ];
 
 const xValueToLabel = {
-  '0.0-1.0': '0',
-  '1.0-10.0': '1-9',
-  '10.0-50.0': '10-49',
-  '50.0-100.0': '50-99',
-  '100.0-250.0': '100-249',
-  '250.0-500.0': '250-499',
-  '500.0-*': '500+',
+  '0--0': '0',
+  '1--9': '1-9',
+  '10--49': '10-49',
+  '50--99': '50-99',
+  '100--249': '100-249',
+  '250--499': '250-499',
+  '500--': '500+',
+};
+
+const typeToColors = {
+  [CITEABLE_BAR_TYPE]: { color: BLUE, hoveredColor: HOVERED_BLUE },
+  [PUBLISHED_BAR_TYPE]: { color: ORANGE, hoveredColor: HOVERED_ORANGE },
 };
 
 class CitationSummaryGraph extends Component {
-  static toSeriesData(bucket) {
+  constructor() {
+    super();
+    this.onBarMouseOut = this.onBarMouseOut.bind(this);
+    this.onCiteableBarHover = this.onCiteableBarHover.bind(this);
+    this.onPublishedBarHover = this.onPublishedBarHover.bind(this);
+    this.onCiteableBarClick = this.onCiteableBarClick.bind(this);
+    this.onPublishedBarClick = this.onPublishedBarClick.bind(this);
+    this.state = {
+      hoveredBar: null,
+    };
+  }
+
+  onCiteableBarClick(datapoint) {
+    this.onBarClick({
+      xValue: datapoint.x,
+      type: CITEABLE_BAR_TYPE,
+    });
+  }
+
+  onPublishedBarClick(datapoint) {
+    this.onBarClick({
+      xValue: datapoint.x,
+      type: PUBLISHED_BAR_TYPE,
+    });
+  }
+
+  onBarClick(clickedBar) {
+    const { onSelectBarChange } = this.props;
+    if (this.isSelectedBar(clickedBar)) {
+      onSelectBarChange(null);
+    } else {
+      onSelectBarChange(clickedBar);
+    }
+  }
+
+  onCiteableBarHover(datapoint) {
+    this.onBarMouseHover({
+      xValue: datapoint.x,
+      type: CITEABLE_BAR_TYPE,
+    });
+  }
+
+  onPublishedBarHover(datapoint) {
+    this.onBarMouseHover({
+      xValue: datapoint.x,
+      type: PUBLISHED_BAR_TYPE,
+    });
+  }
+
+  onBarMouseHover(hoveredBar) {
+    this.setState({ hoveredBar });
+  }
+
+  onBarMouseOut() {
+    this.setState({ hoveredBar: null });
+  }
+
+  getBarColor(bar) {
+    const { selectedBar } = this.props;
+    if (this.isHoveredBar(bar)) {
+      return typeToColors[bar.type].hoveredColor;
+    }
+    if (!selectedBar || this.isSelectedBar(bar)) {
+      return typeToColors[bar.type].color;
+    }
+    return GRAY;
+  }
+
+  toSeriesData(bucket, type) {
     const docCount = bucket.doc_count;
     const countLabel = docCount !== 0 ? docCount.toString() : null;
-    return { x: bucket.key, y: docCount, label: countLabel };
+    return {
+      x: bucket.key,
+      y: docCount,
+      label: countLabel,
+      color: this.getBarColor({ xValue: bucket.key, type }),
+    };
+  }
+
+  isHoveredBar(bar) {
+    const { hoveredBar } = this.state;
+    return (
+      hoveredBar &&
+      hoveredBar.type === bar.type &&
+      hoveredBar.xValue === bar.xValue
+    );
+  }
+
+  isSelectedBar(bar) {
+    const { selectedBar } = this.props;
+    return (
```

Comment:
sorry for the late catch but this and `isHoveredBar` can use `utils/shallowEqual`